pkg/cableengine/syncer/syncer.go

```diff
@@ -201,6 +204,19 @@ func (i *GatewaySyncer) generateGatewayObject() *v1.Gateway {
 	}
 
 	if connections != nil {
+		for index, connection := range *connections {
+			latencySpec, ok := i.latencyMap.Load(connection.Endpoint.Hostname)
+			if ok {
+				latencyInfo := latencySpec.(*healthchecker.LatencyInfo)
+				connection.Latency = latencyInfo.Latency
+				if connection.Status == v1.Connected && latencyInfo.Status != v1.Connected {
+					connection.Status = latencyInfo.Status
+				}
```

Comment:
For the `ConnectionError` status, we should augment it with detail via the `StatusMessage` as we do  elsewhere. I suggest changing the `LatencyInfo` `Status v1.ConnectionStatus` field to `ConnectionError string` and here check:

```suggestion
				if connection.Status == v1.Connected && latencyInfo.ConnectionError != "" {
					connection.Status = vi.ConnectionError
					connection.StatusMessage = latencyInfo.ConnectionError
				}
```

Reply:
Done.